django/forms/formsets.py

```diff
@@ -34,6 +35,7 @@ class ManagementForm(Form):
     def __init__(self, *args, **kwargs):
         self.base_fields[TOTAL_FORM_COUNT] = IntegerField(widget=HiddenInput)
         self.base_fields[INITIAL_FORM_COUNT] = IntegerField(widget=HiddenInput)
+        self.base_fields[EXTRA_FORM_COUNT] = IntegerField(required=False, widget=HiddenInput)
```

Comment:
Do we need the extra field here? Could we do something like this?

`EXTRA_FORM_COUNT  = TOTAL_FORM_COUNT - max(INITIAL_FORM_COUNT, MIN_NUM_FORM_COUNT)` 

(c.f. https://github.com/django/django/blob/d64f192df188903ea5a1ab8dfd6679e5d52aaf9b/django/forms/formsets.py#L117)

Either way `EXTRA_FORM_COUNT ` is purely for client code, yes? So it should go beneath the comment below. The comment should be adjusted, along with [the note on Understanding the management form](https://github.com/django/django/blob/d3b3eb860dc140c56022704672e7f54fa8a0c648/docs/topics/forms/formsets.txt#L262-L269) in the topic docs. 

(Sure we can't calculate it?)

Reply:
`extra` filed is assigned by user. for example:
```python
class ChoiceInline(admin.StackedInline):
    model = Choice
    extra = 3
```
ref: https://docs.djangoproject.com/en/3.1/intro/tutorial07/#writing-your-first-django-app-part-7

I'am sorry that I think `EXTRA_FORM_COUNT = TOTAL_FORM_COUNT - max(INITIAL_FORM_COUNT, MIN_NUM_FORM_COUNT)` is incorrect.

Reply:
Hi @youguanxinqing — but `total_forms` is basically just `extra` + the initial forms, so if we have those two we _shouldn't_ need to set it manually (is my hunch). 

Maybe there's an edge case but I'd like to see that, and have it covered by a test if so. 
I'll have a play with it now, but if you can show how it's wrong, that would be great! 
(Thanks! 👍)

Reply:
Sorry for my mistabke, you are right.